yt_dlp/postprocessor/ffmpeg.py

```diff
@@ -619,13 +619,19 @@ def run(self, info):
         webm_vtt_warn = False
         mp4_ass_warn = False
 
+        json_names, json_filenames = [], []
+
         for lang, sub_info in subtitles.items():
             if not os.path.exists(sub_info.get('filepath', '')):
                 self.report_warning(f'Skipping embedding {lang} subtitle because the file is missing')
                 continue
             sub_ext = sub_info['ext']
             if sub_ext == 'json':
-                self.report_warning('JSON subtitles cannot be embedded')
+                if info['ext'] in ('mkv', 'mka'):
+                    json_names.append(lang)
+                    json_filenames.append(sub_info['filepath'])
```

Comment:
```suggestion
                    json_subs[lang] = ]sub_info['filepath']
```
is cleaner

Set `json_subs = {}` above and use `lang, filename in json_subs.items()`  below